### PR TITLE
(WIP) Display Fx Icon in timeline/xsheet by manongjohn

### PR DESCRIPTION
**Pull request by JPUser1**
Display Fx Icon in Tahoma2D to update to opentoon_Xsheet thumbnail

### CODE REVIEW

**Comment by manongjohn** [comment]
This appears to be a port of my code from Tahoma. Please give credit where credit is due.

File: toonz/sources/toonz/xshcolumnviewer.cpp [hunk 1]
```diff
@@ -1092,6 +1082,8 @@ void ColumnArea::DrawHeader::drawThumbnail(QPixmap &iconPixmap) const {
         levelColumn->setIconVisible(true);
       else if (meshColumn)
         meshColumn->setIconVisible(true);
+      else if (zColumn)
+        zColumn->setIconVisible(true);
     }
```
Comment (manongjohn) [inline toonz/sources/toonz/xshcolumnviewer.cpp:1087]: You appear to have an extra '}' that is causing this build to fail.

**Comment by shun-iwasawa** [comment]
@JPUser1 

Co-authored commit title should be like this:
```
Display Fx Icon as timeline/xsheet by manongjohn


Co-authored-by: manongjohn <manongjohn@users.noreply.github.com>
```
Please see the [GitHub document](https://docs.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors) for details.


Since this PR is a feature addition, it will not be merged until releasing of the next version. Thank you for your patience!

**Comment by RodneyBaker** [comment]
As this PR will not be merged until after release of v1.5 it would be good to discuss the actual icons that get displayed with this feature.   Currently they seem to be a bit random and not optimized.  

By not optimized I mean to say they don't fully fill the space provided for the icon.  
The randomness of the images themselves might be a bit difficult to overcome as each attempts to display an example of the fx that will be generated.  My initial thought was that perhaps they could all be grayscale or of a color scheme that complements each other.

**Comment by RodneyBaker** [comment]
There are two primary considerations that still appear to be open regarding this PR:
1.  Whether this feature should be optional
2.  If the current icons should be updated

Neither of these two are a direct impediment to merging this port from Tahoma but should be considered along with any other concerns (if any).   As such we welcome any and all feedback.

Changing from [Do Not Merge] to (WIP).

**Comment by RodneyBaker** [comment]
Merging this as preview/icon updates can follow up later should they be desireable.

**Pull request merged**